app/photosketchingcontroller.h

```diff
@@ -0,0 +1,102 @@
+/***************************************************************************
+*                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef PHOTODRAWINGCONTROLLER_H
+#define PHOTODRAWINGCONTROLLER_H
+
+#include <QPointF>
+#include <QObject>
+#include <QColor>
+#include <QtQml/qqmlregistration.h>
+
+
+/**
+ * Utility class, which is used in place of std::pair, because of QML limitations.
+ */
+class ColorPath
+{
+    Q_GADGET
+    Q_PROPERTY( QColor color MEMBER mColor )
+    Q_PROPERTY( QVector<QPointF> points MEMBER mPoints )
+  public:
+    ColorPath( const QColor color, const QVector<QPointF> &list ): mColor( color ), mPoints( list ) {};
+    ColorPath();
+
+    bool operator==( const ColorPath &other ) const;
+
+    QColor mColor;
+    QVector<QPointF> mPoints;
+};
+
+class PhotoSketchingController: public QObject
+{
+    Q_OBJECT
+    QML_ELEMENT
+
+    Q_PROPERTY( QString photoSource MEMBER mPhotoSource REQUIRED );
+    Q_PROPERTY( bool canUndo MEMBER mCanUndo NOTIFY canUndoChanged );
+    Q_PROPERTY( QColor activeColor MEMBER mPenColor WRITE setActiveColor NOTIFY activeColorChanged );
+    Q_PROPERTY( double photoScaleRatio MEMBER mPhotoScaleRatio WRITE setPhotoScaleRatio NOTIFY photoScaleRatioChanged );
+
+  public:
+    explicit PhotoSketchingController( QObject *parent = nullptr ): QObject( parent ) {};
+    ~PhotoSketchingController() override = default;
+
+    // starts new line, to store points in
+    Q_INVOKABLE void newSketch();
+
+    // adds new point to current line
+    Q_INVOKABLE void addPoint( QPointF newPoint );
+
+    // sets color of new line
+    Q_INVOKABLE void setActiveColor( QColor newColor );
```

Comment:
This could be const ref, right?
```suggestion
    Q_INVOKABLE void setActiveColor( const QColor &newColor );
```